src/core/qgsapplication.h

```diff
@@ -1134,6 +1141,7 @@ class CORE_EXPORT QgsApplication : public QApplication
       QgsProcessingRegistry *mProcessingRegistry = nullptr;
       QgsConnectionRegistry *mConnectionRegistry = nullptr;
       std::unique_ptr<QgsProjectStorageRegistry> mProjectStorageRegistry;
+      std::unique_ptr<QgsLayerMetadataProviderRegistry> mLayerMetadataProviderRegistry;
```

Comment:
This should be a raw pointer, with a manual delete in ~ApplicationMembers() in the correct place for clean tear-down. (same with mProjectStorageRegistry, but don't feel you need to fix that here)

Reply:
Fixed both.